internal/server/storage/pg_multiput.go

```diff
@@ -0,0 +1,149 @@
+package humaystorage
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/sethvargo/go-retry"
+)
+
+const (
+	insertQuery = `
+	INSERT INTO %s
+    (name, value)
+	VALUES %s;
```

Comment:
```suggestion
	VALUES %s
	ON CONFLICT(name) DO UPDATE SET value = EXCLUDED.value;
```
и не надо будет городить огород с предварительным чтением и выбором метода сохранения.
https://www.dbvis.com/thetable/postgresql-upsert-insert-on-conflict-guide/